candle-core/src/tensor.rs

```diff
@@ -1235,6 +1235,74 @@ impl Tensor {
         Ok(from_storage(storage, (n, c, h_out, w_out), op, false))
     }
 
+    /// Computes the dot product between two tensors
```

Comment:
The pytorch version only supports 1d tensor and mention doing this intentiionally so I would suggest doing the same here as we try to be as close as possible as their api.
https://docs.pytorch.org/docs/stable/generated/torch.dot.html

Reply:
Ok.